addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -111,6 +111,12 @@ options.registry.TableOfContent = options.Class.extend({
         const exception = (tocEl) => !tocEl.querySelector('.s_table_of_content_navbar a');
         this._activateScrollSpy(exception);
         this.oldHeadingsEls = [...headingsEls.map(el => el.cloneNode(true))];
+        const desktopInvisibleEls = targetedEls.filter(el => el.closest('.o_snippet_desktop_invisible'));
+        if (this.oldHeadingsEls.length === 0 && desktopInvisibleEls.length === 0) {
+            // destroy public widget and remove the ToC since there are no more
+            // child elements.
+            this.trigger_up('remove_snippet', {$snippet: this.$target});
```

Comment:
There is something I don't understand here... We shouldn't worry about the desktop/mobile visibility of the elements that make up the ToC no ?

Reply:
Hmm...actually the thing is we're handling the removal of the ToC block based on the Heading Elements`headingsEls` so if 
we don't consider the device(desktop) visibility and if you try to make the whole ToC snippet invisible we're not getting any `headingsEls` hence instead of making ToC invisible for a device, it will remove the Snippet.

Reply:
Ok but even if an element is invisible, it will be in `targetedEls` right ?
So why can't we do
```js
if (targetedEls.length === 0) {
    // destroy public widget and remove the ToC since there are no more
    // child elements.
    this.trigger_up('remove_snippet', {$snippet: this.$target});
```

Reply:
ohh...yes surely we can, in the end, that's what we're focusing on for ToC.